### PR TITLE
Create page description institution on mobile

### PR DESCRIPTION
**Pull request by MaianaB**
**Feature/Bug description:** Doesn't exist the description institution page.

**Solution:** Create according to the new design
.
**TODO/FIXME:** n/a

captura de tela de 2019-02-18 10-44-46
captura de tela de 2019-02-18 10-43-47
captura de tela de 2019-02-18 10-43-33
captura de tela de 2019-02-18 10-43-24

### CODE REVIEW

File: frontend/app.js [hunk 1]
```diff
@@ -599,6 +611,6 @@
     }
 
     (function main() {
-        initServiceWorker();
+        //initServiceWorker();
```
Comment (JuliePessoa) [inline frontend/app.js]: undo

File: frontend/app.js [hunk 2]
```diff
@@ -164,6 +164,18 @@
                     }
                 }
             })
+            .state(STATES.INST_DESCRIPTION, {
```
Comment (JuliePessoa) [inline frontend/app.js:167]: add this state in the interceptor for only mobile screens

File: frontend/app.js (hunk 1 above)
Comment (Luiz-FS) [inline frontend/app.js]: Undo

File: frontend/institution/descriptionInst/descriptionController.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
+(function(){
+    var app = angular.module('app');
```
Comment (JuliePessoa) [inline frontend/institution/descriptionInst/descriptionController.js]: const

File: frontend/institution/descriptionInst/descriptionController.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        descriptionCtrl.$onInit = () => {
+            descriptionCtrl.institution = institution;
+            descriptionCtrl.currentInstitutionKey = institutionKey;
+            observer = ObserverRecorderService.register(descriptionCtrl.institution);
+        };
+
+        /** Save changes of institution and emit event. 
+         */
+        descriptionCtrl.save = () => {
+            var patch = ObserverRecorderService.generate(observer);
```
Comment (JuliePessoa) [inline frontend/institution/descriptionInst/descriptionController.js]: const

File: frontend/institution/descriptionInst/descriptionController.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+            InstitutionService.update(descriptionCtrl.currentInstitutionKey, patch).then(
+                function success() {
+                   $rootScope.$emit('EDIT_DESCRIPTION_INST');
+                   $mdDialog.hide();
+                }
+            );
+        }    
+    });
+    
+    app.controller("DescriptionInstController", function DescriptionInstController($state, InstitutionService, $rootScope){
```
Comment (JuliePessoa) [inline frontend/institution/descriptionInst/descriptionController.js:4]: create another file for this controller.

File: frontend/institution/descriptionInst/descriptionController.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+        const descriptionCtrl = this;
+        descriptionCtrl.isLoading = false;
+
+        descriptionCtrl.$onInit = () => {
+            descriptionCtrl.isLoading = true;        
+            InstitutionService.getInstitution($state.params.institutionKey).then(function(institution){
+                descriptionCtrl.institution = institution;
+                descriptionCtrl.isLoading = false;
+
```
Comment (JuliePessoa) [inline frontend/institution/descriptionInst/descriptionController.js]: remove blank line

File: frontend/institution/institutionController.js [hunk 7]
```diff
@@ -263,6 +269,20 @@
             });
         };
 
+        institutionCtrl.editDescription = function(ev) {
+            $mdDialog.show({
+                templateUrl: 'app/institution/descriptionInst/edit_description.html',
+                targetEvent: ev,
+                clickOutsideToClose:true,
+                locals: {
+                    institution: institutionCtrl.institution,
```
Comment (JuliePessoa) [inline frontend/institution/institutionController.js]: why do you need both institution and institution key as parameters? Doenst institution object have it own key property?

**Review by JuliePessoa — CHANGES_REQUESTED** [review]

**Review by Luiz-FS — CHANGES_REQUESTED** [review]

File: frontend/institution/descriptionInst/descriptionController.js [hunk 8]
```diff
@@ -0,0 +1,51 @@
+(function(){
+    var app = angular.module('app');
+
+    app.controller("EditDescriptionController", function EditDescriptionController(institution, institutionKey, InstitutionService,
```
Comment (Luiz-FS) [inline frontend/institution/descriptionInst/descriptionController.js]: Test

**Review by Luiz-FS — CHANGES_REQUESTED** [review]

File: frontend/institution/descriptionInst/descriptionController.js [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+
+        descriptionCtrl.$onInit = () => {
+            descriptionCtrl.isLoading = true;        
+            InstitutionService.getInstitution($state.params.institutionKey).then(function(institution){
+                descriptionCtrl.institution = institution;
+                descriptionCtrl.isLoading = false;
+            })
+        };
+
+        /** Listenner edit description inst event, and should refrash institution.
```
Comment (raonismaneoto) [inline frontend/institution/descriptionInst/descriptionController.js]: refresh*

File: frontend/institution/descriptionInst/editDescriptionController.js [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+            observer = ObserverRecorderService.register(descriptionCtrl.institution);
+        };
+
+        /** Save changes of institution and emit event. 
+         */
+        descriptionCtrl.save = () => {
+            const patch = ObserverRecorderService.generate(observer);
+            InstitutionService.update(descriptionCtrl.institution.key, patch).then(
+                function success() {
+                    console.log("-------------------------")
```
Comment (Luiz-FS) [inline frontend/institution/descriptionInst/editDescriptionController.js]: Remove console

File: frontend/institution/descriptionInst/editDescriptionController.js [hunk 11]
```diff
@@ -0,0 +1,27 @@
+(function(){
+    const app = angular.module('app');
+
+    app.controller("EditDescriptionController", function EditDescriptionController(institution, InstitutionService,
```
Comment (Luiz-FS) [inline frontend/institution/descriptionInst/editDescriptionController.js]: Use inline annotation

File: frontend/institution/descriptionInst/editDescriptionController.js [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+    
+        descriptionCtrl.$onInit = () => {
+            descriptionCtrl.institution = institution;
+            observer = ObserverRecorderService.register(descriptionCtrl.institution);
+        };
+
+        /** Save changes of institution and emit event. 
+         */
+        descriptionCtrl.save = () => {
+            const patch = ObserverRecorderService.generate(observer);
```
Comment (raonismaneoto) [inline frontend/institution/descriptionInst/editDescriptionController.js]: I'm not sure about how this service works, but shouldn't u unregiste the observer after that? Maybe this observer leaves even when this controller dies...

File: frontend/institution/descriptionInst/editDescriptionController.js (hunk 10 above)
Comment (raonismaneoto) [inline frontend/institution/descriptionInst/editDescriptionController.js]: remove it

**Review by Luiz-FS — CHANGES_REQUESTED** [review]

File: frontend/institution/institutionHeader.component.js [hunk 13]
```diff
@@ -33,6 +33,13 @@
                 return $state.current.name == STATES.INST_REGISTRATION_DATA;
             }
 
+            /** Return if current state is registration data on institution.
+             * 
+             */
+            instHeaderCtrl.isDescription = function isDescription(){
+                return $state.current.name == STATES.INST_DESCRIPTION;
+            }
```
Comment (raonismaneoto) [inline frontend/institution/institutionHeader.component.js]: remember to use semicolons all over this file

File: frontend/institution/base_institution.css [hunk 14]
```diff
@@ -19,6 +19,12 @@
     padding: 5px;
 }
 
+.base-content-inst{
+    height: 100%;
+    position: absolute;
```
Comment (pedroespindula) [inline frontend/institution/base_institution.css:24]: Why use absolute?

File: frontend/utils/hideNavbar.js [hunk 15]
```diff
@@ -12,33 +12,47 @@
...
+                ];
                 
                 /** Definy initial style of toolbars according the current state.
                  */
                 scope.initialToolbarDisplayState = function initialToolbarDisplayState(){
-                    const shouldHideBottomToolbar = !scope.isBottomToolbarAllowed() || 
-                        scope.INSTITUTION_STATES.includes($state.current.name);
-                    if (!scope.isStateAllowedTopMobile) scope.hideElement(scope.topTollbar);
-                    if (shouldHideBottomToolbar) scope.hideElement(scope.bottomToolbar);
+                    if(Utils.isMobileScreen(450)){
```
Comment (raonismaneoto) [inline frontend/utils/hideNavbar.js:35]: use SCREEN_SIZES constant

**Review by raonismaneoto — CHANGES_REQUESTED** [review]

File: frontend/institution/descriptionInst/description_inst.css [hunk 16]
```diff
@@ -0,0 +1,53 @@
+#textarea-edit-description{
+    height: 20em !important;
```
Comment (pedroespindula) [inline frontend/institution/descriptionInst/description_inst.css]: `!important` is really necessary?

**Review by pedroespindula — CHANGES_REQUESTED** [review]
Edit institution description has scroll

File: frontend/institution/descriptionInst/description_inst.css (hunk 16 above)
Comment (MaianaB) [reply frontend/institution/descriptionInst/description_inst.css]: Yes, the material library is overwriting this property

File: frontend/institution/descriptionInst/editDescriptionController.js [hunk 17]
```diff
@@ -1,27 +1,29 @@
...
             descriptionCtrl.institution = institution;
-            observer = ObserverRecorderService.register(descriptionCtrl.institution);
+            descriptionCtrl.institutionClone = _.cloneDeep(institution);
         };
 
         /** Save changes of institution and emit event. 
          */
         descriptionCtrl.save = () => {
-            const patch = ObserverRecorderService.generate(observer);
+            let clone = JSON.parse(angular.toJson(descriptionCtrl.institutionClone));
```
Comment (raonismaneoto) [inline frontend/institution/descriptionInst/editDescriptionController.js]: const

File: frontend/institution/descriptionInst/editDescriptionController.js [hunk 18]
```diff
@@ -1,27 +1,29 @@
...
+            let modified = JSON.parse(angular.toJson(descriptionCtrl.institution));
```
Comment (raonismaneoto) [inline frontend/institution/descriptionInst/editDescriptionController.js]: const

**Review by raonismaneoto — CHANGES_REQUESTED** [review]

File: frontend/app.js [hunk 19]
```diff
@@ -612,6 +612,6 @@
     }
 
     (function main() {
-        initServiceWorker();
+        //initServiceWorker();
```
Comment (raonismaneoto) [inline frontend/app.js]: undo

**Review by raonismaneoto — CHANGES_REQUESTED** [review]

**Review by Luiz-FS — CHANGES_REQUESTED** [review]
Broken on small screens

**Review by raonismaneoto — APPROVED** [review]

**Review by JuliePessoa — APPROVED** [review]

**Review by pedroespindula — APPROVED** [review]

**Review by tiago-lp — APPROVED** [review]

**Review by Luiz-FS — APPROVED** [review]

**Pull request merged**